File: mifosng-android/src/main/java/com/mifos/api/model/Payload.java
/*
 * This project is licensed under the open source MPL V2.
 * See https://github.com/openMF/android-client/blob/master/LICENSE.md
 */

package com.mifos.api.model;

//TODO Remove calendarId and TransactionDate from this Payload class;
public class Payload {
    private String dateFormat = "dd MMMM YYYY";
    private String locale = "en";
    private long calendarId;
    private String transactionDate;

    public String getDateFormat() {
        return dateFormat;
    }

    public void setDateFormat(String dateFormat) {
        this.dateFormat = dateFormat;
    }

    public String getLocale() {
        return locale;
    }

    public void setLocale(String locale) {
        this.locale = locale;
    }

    public long getCalendarId() {
        return calendarId;
    }

    public void setCalendarId(long calendarId) {
        this.calendarId = calendarId;
    }

    public String getTransactionDate() {
        return transactionDate;
    }

    public void setTransactionDate(String transactionDate) {
        this.transactionDate = transactionDate;
    }

    @Override
    public String toString() {
        return "{" +
                "dateFormat='" + dateFormat + '\'' +
                ", locale='" + locale + '\'' +
                ", calendarId=" + calendarId +
                ", transactionDate='" + transactionDate + '\'' +
                '}';
    }
}

File: core/network/src/main/java/com/mifos/core/network/model/Payload.kt
/*
 * Copyright 2024 Mifos Initiative
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at https://mozilla.org/MPL/2.0/.
 *
 * See https://github.com/openMF/android-client/blob/master/LICENSE.md
 */
package com.mifos.core.network.model

// TODO Remove calendarId and TransactionDate from this Payload class;
open class Payload {
    var dateFormat = "dd MMMM YYYY"
    var locale = "en"
    var calendarId: Long = 0
    var transactionDate: String? = null
    override fun toString(): String {
        return "{" +
            "dateFormat='" + dateFormat + '\'' +
            ", locale='" + locale + '\'' +
            ", calendarId=" + calendarId +
            ", transactionDate='" + transactionDate + '\'' +
            '}'
    }
}
